public/app/core/components/TransformersUI/FilterByValueTransformerEditor.tsx

```diff
@@ -0,0 +1,231 @@
+          }}
+        />
+      </>
+    );
+  }
+
+  return (
+    <div className="gf-form-inline">
+      <div className="gf-form gf-form-spacing">
+        <div className="gf-form-label width-8">Filter Type</div>
+        <Select
+          className="width-8"
+          options={[
+            { label: 'Include', value: 'include' },
+            { label: 'Exclude', value: 'exclude' },
+          ]}
+          value={config.type}
+          onChange={option => {
+            // console.log('onChange filterType', option.value);
+            onConfigChange({ ...config, type: option.value });
+          }}
+          menuPlacement="bottom"
+        />
+      </div>
+      <div className="gf-form gf-form-spacing">
+        <div className="gf-form-label width-8">Filter on Field</div>
+        <Select
+          className="width-16"
+          placeholder="Field Name"
+          options={fieldNameOptions}
+          value={config.fieldName}
+          invalid={fieldNameInvalid}
+          onChange={value => {
+            // console.log('onChange fieldName', value);
+            if (value === null) {
+              onConfigChange({ ...config, fieldName: null });
+            } else {
+              onConfigChange({ ...config, fieldName: value.value });
+            }
+          }}
+          isClearable
+          menuPlacement="bottom"
+        />
+      </div>
+      <div className="gf-form gf-form-spacing">
+        <div className="gf-form-label width-4">Test</div>
+        <Select
+          invalid={filterTypeInvalid}
+          className="width-8"
+          placeholder="Select test"
+          options={filterTypeOptions}
+          value={config.filterType}
+          onChange={value => {
+            // console.log('onChange test', value);
+            onConfigChange({ ...config, filterType: value.value });
+          }}
+          menuPlacement="bottom"
+        />
+      </div>
+      <div className="gf-form gf-form--grow gf-form-spacing ">{filterOptionsInput}</div>
+      <div className="gf-form">
+        <Button icon="times" onClick={onDelete} style={{ height: '100%' }} size="sm" variant="secondary" />
+      </div>
+    </div>
+  );
+}
+
+export const FilterByValueTransformerEditor: React.FC<TransformerUIProps<FilterByValueTransformerOptions>> = ({
+  input,
+  options,
+  onChange,
+}) => {
+  const fieldNames = useMemo(() => getAllFieldNamesFromDataFrames(input), [input]);
+  const fieldNameOptions = fieldNames.map((item: string) => ({ label: item, value: item }));
+
+  const onAddFilter = () => {
+    let valueFilters = options.valueFilters.map(filter => ({ ...filter })); // Deep copy
+    valueFilters.push({
+      type: 'include',
+      fieldName: null,
+      filterExpression: null,
+      filterType: ValueFilterID.regex,
+    });
+
+    onChange({
+      ...options,
+      valueFilters,
+    });
+  };
+
+  const onDeleteFilter = (index: number) => () => {
```

Comment:
All of these `callback` functions that are defined in the render function should be wrapped in a [useCallback](https://reactjs.org/docs/hooks-reference.html#usecallback) react-hook to prevent being created on each render.